### ISSUE TITLE
[BUG] OllamaStreamingLanguageModel throws EOFException

### ISSUE THREAD

**Issue by bbrynski** [langchain4j/langchain4j#1750]
**Describe the bug**

Using StreamingLanguageModel to generate response from model and when response from model is short then is working fine. But when model response a lot of it throws: `com.google.gson.JsonSyntaxException: java.io.EOFException: End of input at line 1 column 1025 path $.context[0]`
I using local model: llama2, exception throws around 1 minute after generating response.

**Log and Stack trace**

```
com.google.gson.JsonSyntaxException: java.io.EOFException: End of input at line 1 column 1025 path $.context[0]
        at com.google.gson.Gson.fromJson(Gson.java:1370)
        at com.google.gson.Gson.fromJson(Gson.java:1262)                                                                                                                                                                                
        at com.google.gson.Gson.fromJson(Gson.java:1171)                                                                                                                                                                                
        at com.google.gson.Gson.fromJson(Gson.java:1107)                                                                                                                                                                                
        at dev.langchain4j.model.ollama.OllamaClient$1.onResponse(OllamaClient.java:120)                                                                                                                                                
        at retrofit2.OkHttpCall$1.onResponse(OkHttpCall.java:161)                                                                                                                                                                       
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)                                                                                                                                                          
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)                                                                                                                                    
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)                                                                                                                                    
        at java.base/java.lang.Thread.run(Thread.java:840)                                                                                                                                                                              
Caused by: java.io.EOFException: End of input at line 1 column 1025 path $.context[0]                                                                                                                                                   
        at com.google.gson.stream.JsonReader.nextNonWhitespace(JsonReader.java:1542)                                                                                                                                                    
        at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:550)                                                                                                                                                                
        at com.google.gson.stream.JsonReader.skipValue(JsonReader.java:1339)                                                                                                                                                            
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:514)                                                                                                               
        at com.google.gson.Gson.fromJson(Gson.java:1361)                                                                                                                                                                                
        ... 9 more     
```
**To Reproduce**
```
<!-- Please provide a relevant code snippets to reproduce this bug. -->
 StreamingLanguageModel ollama = OllamaStreamingLanguageModel.builder()
            .baseUrl("http://localhost:11434/")
            .modelName("llama2")
            .build();

CompletableFuture<String> future = new CompletableFuture<>();
ollama.generate("why the sky is blue?", new StreamingResponseHandler<String>() {

            @Override
            public void onNext(String token) {
                System.out.println(token);
            }

            @Override
            public void onComplete(dev.langchain4j.model.output.Response<String> response) {
                future.complete(response.toString());
            }

            @Override
            public void onError(Throwable error) {
                error.printStackTrace();
            }
        });
        future.join();
```
**Expected behavior**

No exception is thrown when processing streaming response.

**Please complete the following information:**
- LangChain4j version: 0.32.0
- LLM(s) used: Ollama llama2
- Java version: 17

**Comment by langchain4j** [comment]
@bbrynski please take a look at https://github.com/langchain4j/langchain4j/pull/867, it can be ported to `OllamaClient.streamingCompletion()`

**Comment by langchain4j** [comment]
BTW is there a reason why you are using `OllamaStreamingLanguageModel` instead of `OllamaStreamingChatModel`?

**Comment by bbrynski** [comment]
I using `OllamaStreamingLanguageModel ` because i want to use `/api/generate` endpoint
Regarding that #867 is looks as similiar issue

**Comment by langchain4j** [comment]
@bbrynski would you like to do the same fix as in https://github.com/langchain4j/langchain4j/pull/867 for `OllamaClient.streamingCompletion()` ?

**Comment by Martin7-1** [comment]
@langchain4j @bbrynski Maybe I can fix it when I develop #1756

**Comment by bbrynski** [comment]
@Martin7-1 if you could that will be great, thanks
